Source/core/Time.cpp

```diff
@@ -639,19 +639,23 @@ namespace Core {
 
     Time::Time(const uint16_t year, const uint8_t month, const uint8_t day, const uint8_t hour, const uint8_t minute, const uint8_t second, const uint16_t millisecond, const bool localTime)
     {
-        _time.wYear = year;
-        _time.wMonth = month;
-        _time.wDay = day;
-        _time.wHour = hour % 24;
-        _time.wMinute = minute % 60;
-        _time.wSecond = second % 60;
-        _time.wMilliseconds = millisecond;
-        _time.wDayOfWeek = static_cast<WORD>(~0);
-
-        if (localTime) {
-            SYSTEMTIME convertedTime;
-            TzSpecificLocalTimeToSystemTime(nullptr, &_time, &convertedTime);
-            _time = convertedTime;
+        if ((year >= 1900) && ((month >= 1) && (month <= 12)) && ((day >= 1 ) && (day <= 31)) &&
+            (hour <= 24) && (minute <= 60) && (second <= 60) && (millisecond <= 999)) {
+
+            _time.wYear = year;
+            _time.wMonth = month;
+            _time.wDay = day;
+            _time.wHour = hour % 24;
+            _time.wMinute = minute % 60;
+            _time.wSecond = second % 60;
+            _time.wMilliseconds = millisecond;
+            _time.wDayOfWeek = static_cast<WORD>(~0);
+
+            if (localTime) {
+                SYSTEMTIME convertedTime;
+                TzSpecificLocalTimeToSystemTime(nullptr, &_time, &convertedTime);
+                _time = convertedTime;
+            }
```

Comment:
as _time is not set initialized in the constructor initializer list should we do that or have an else here for the IsValid to work?

Reply:
Initialise with some default values you mean ? Currently it is working with default values. But as you suggested, I think better to init with default to avoid garbage as well

Reply:
I found initialisation of _time is already there in the code, expect with this windows based contractor. Added here as well